src/libs/blueprint/conduit_blueprint_mesh.cpp

```diff
@@ -3707,26 +4015,62 @@ mesh::topology::unstructured::generate_corners(const Node &topo,
             if(is_topo_3d)
             {
                 corner_face_count = corner_entities.size();
-                conn_data_raw.push_back(corner_entities.size());
+                size_data_raw.push_back(corner_entities.size());
+
             }
 
             for(index_t bfi = 0; bfi < corner_face_count; bfi++)
             {
-                conn_data_raw.push_back(corners_face_degree);
+                std::vector<int64> face;
                 for(index_t fi = 0; fi < corners_face_degree - 1; fi++)
                 {
                     index_t fei = ((bfi + fi) % corner_entities.size());
                     index_t corner_face_index = corner_entities[fei];
                     index_t corner_index_dim = (fei % 2 == 0) ?
                         line_shape.dim : face_shape.dim;
-                    conn_data_raw.push_back(
+                    face.push_back(
                         dim_coord_offsets[corner_index_dim] + corner_face_index);
                 }
 
                 index_t corner_cap_index = ((bfi % 2) == 0) ?
                     dim_coord_offsets[point_shape.dim] + point_index :
                     dim_coord_offsets[topo_shape.dim] + elem_index;
-                conn_data_raw.push_back(corner_cap_index);
+                face.push_back(corner_cap_index);
+
+                if (is_topo_3d)
+                {
+                    bool has_perm = false;
+                    index_t face_index = subconn_data_raw.size() / corners_face_degree;
+
+                    // Check if face already exists
+                    for (index_t face_i = 0; face_i < face_index; face_i++ )
+                    {
+                        index_t face_off = face_i * corners_face_degree;
+                        has_perm |= std::is_permutation(subconn_data_raw.begin() + face_off,
+                                                        subconn_data_raw.begin() + face_off + 
+                                                        corners_face_degree,
+                                                        face.begin());
```

Comment:
Very cool; I didn't know `std::is_permutation` even existed!

Reply:
that is cool, it's c++11 and forward. We still have an option for older C++, but I think for the next release we should make C++11 support min